apps/api/src/models/ApplicationData.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+from pydantic import BaseModel, EmailStr, Field, HttpUrl
+
+from .utils import form_body
+
+
+class Decision(str, Enum):
+    ACCEPTED = "ACCEPTED"
+    WAITLISTED = "WAITLISTED"
+    REJECTED = "REJECTED"
+
+
+Review = tuple[datetime, str, Decision]
+
+
+@form_body
+class RawApplicationData(BaseModel):
+    first_name: str
+    last_name: str
+    email: EmailStr
+    pronouns: list[str]
+    ethnicity: str
+    is_18_older: bool
+    university: str
+    education_level: str
+    major: str
+    is_first_hackathon: bool
+    portfolio_link: Optional[HttpUrl]
+    linkedin_link: Optional[HttpUrl]
+    collaboration_question: Optional[str] = Field()
+    any_job_question: str = Field()
+
+    class Config:
+        anystr_strip_whitespace = True
+        max_anystr_length = 254
```

Comment:
I think the max length in the config will override any individual properties although i'm not sure if specifying `Field` affects that. See HackAtUCI/HackUCI-Site#30 for the original discussion.

Reply:
I did some testing on my own machine with `RawApplicationData` and it turns out that the properties in `Field` override those in `ConfigDict`. For example, if the `Field` has `max_length = 10` and `ConfigDict` has `str_max_length = 254`, Pydantic will raise an exception if the provided string is greater than 10 characters.